share/commands/technique-migrate

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+. "${BASEDIR}/../lib/api_call.sh"
+. "${BASEDIR}/../lib/common.sh"
+
+TOKEN="$(cat /var/rudder/run/api-token)"
+DISPLAY_COMMAND=false
+
+while getopts "iIvdqcn:" opt; do
+  case $opt in
+    i|I|v|d)
+      DISPLAY_COMMAND=true
+      ;;
+    n)
+      TECHNIQUE_NAME="${OPTARG}"
+      ;;
+    c)
+      clear_colors
+      ;;
+  esac
+done
+# to keep the argument as $1
+shift  $((OPTIND-1))
+
+if [ -z "${TECHNIQUE_NAME}" ]
+then
+  echo "Usage: rudder technique migrate [-iIcvd] -n technique_name [target_technique_version]"
+  exit 1
+fi
+
+TECHNIQUE_VERSION="$1"
+if [ -z "${TECHNIQUE_VERSION}" ]
+then
+  TECHNIQUE_VERSION="latest"
+fi
+
+if [ "${TECHNIQUE_VERSION}" == "latest" ]
+then
+# Find the newest technique version
+  TECHNIQUE_VERSION=$(complete_api_call "${API_URL}/api/latest/techniques" "${TOKEN}" "GET" "| jq -r '.data | .techniques[] | select( .name == \"${TECHNIQUE_NAME}\") | .versions[]' | tail -n1" "${DISPLAY_COMMAND}")
+  if [ -z "${TECHNIQUE_VERSION}" ];
+  then
+    printf "${RED}No version found for the technique ${TECHNIQUE_NAME}${NORMAL}\n"
+    exit 1
+  fi
+fi
+
+echo "TECHNIQUE ${TECHNIQUE_NAME} will migrate to ${TECHNIQUE_VERSION}"
+echo ""
+
+# Find the directives using the given technique
+DIRECTIVES=$(complete_api_call "${API_URL}/api/latest/directives" "${TOKEN}" "GET" "| jq -r '.data.directives[] | select(.techniqueName==\"${TECHNIQUE_NAME}\") | .id' | sed 's/\"//g'" "${DISPLAY_COMMAND}" )
```

Comment:
You should not pass a pipe as a string to a function to eval it
you should call the function and pass the result directly to the pipe

Reply:
if I do this we will no longer be able to print the exact command executed easily since one part will be local and the other one will come from the api_call.sh script.